### PR TITLE
Prototype caching

### PR DESCRIPTION
**Pull request by Feasoron**
Created a caching system for edge. Supports Redis and Application caching, with everything defaulting to the application cache. Currently caches Dynamic Modules.

### CODE REVIEW

**Comment by frxnz** [comment]
@Feasoron Can we we review this in the meeting tomorrow?

**Comment by Feasoron** [comment]
Sure. I was planning on going over it in the meeting, if you want to do that via this pull request I'm fine with it. Unless you are planning on being mean and making a public mocking of my code, in which case I'll cry and you'll feel bad.

File: DirigoEdgeCore/Packaging/DirigoEdgeCore.nuspec [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <package >
   <metadata>
     <id>DirigoEdgeCore</id>
-    <version>1.2.0</version>
+    <version>1.2.1</version>
```
Comment (frxnz) [inline DirigoEdgeCore/Packaging/DirigoEdgeCore.nuspec]: This should be 1.2.2, I'm not sure why 1.2.0 is the older version. I would wait until we merge this to change the version though.

**Comment by frxnz** [comment]
This is a safe place. Also I don't have feelings.

File: DirigoEdgeCore/Utils/DynamicModules.cs [hunk 2]
```diff
@@ -24,6 +25,46 @@ private DynamicModules()
...
+
+        public String GetHtmlByShortCode(ShortCodeObject shortCode)
+        {
+
+            return !Instance.GetDynamicModuleList().ContainsKey(shortCode.Name)
+                ? String.Empty
+                : Instance.GetDynamicModuleList()[shortCode.Name].GetHtml(shortCode.Parameters);
+        }
+
+        public String GeCssByShortCode(String shortCode)
```
Comment (frxnz) [inline DirigoEdgeCore/Utils/DynamicModules.cs]: Should be `GetCssByShortCode`

**Comment by Feasoron** [comment]
Getting the Section Listing in a vanilla 2.0 isn't going to happen - it needs so much ski and other things that are different that it's a significant undertaking. We'll have to add this to the list of things to verify when we port something using it to 2.0

**Pull request merged**